src/KeyCommand.web.js

```diff
@@ -0,0 +1,142 @@
+ *
+ * @param {string} type - Event name.
+ * @param {Function} callback - Callback to be called when the event is triggered.
+ * @returns {Object}
+ */
+EventEmitter.addListener = (type, callback) => {
+    EventEmitter.on(type, callback);
+    return {
+        remove: () => EventEmitter.removeListener(type, callback),
+    };
+};
+
+const commands = [];
+const constants = {
+    keyInputDownArrow: 'keyInputDownArrow',
+    keyInputEscape: 'keyInputEscape',
+    keyInputLeftArrow: 'keyInputLeftArrow',
+    keyInputRightArrow: 'keyInputRightArrow',
+    keyInputUpArrow: 'keyInputUpArrow',
+    keyModifierCapsLock: 'keyModifierCapsLock',
+    keyModifierCommand: 'keyModifierCommand',
+    keyModifierControl: 'keyModifierControl',
+    keyModifierControlCommand: 'keyModifierControlCommand',
+    keyModifierControlOption: 'keyModifierControlOption',
+    keyModifierControlOptionCommand: 'keyModifierControlOptionCommand',
+    keyModifierNumericPad: 'keyModifierNumericPad',
+    keyModifierOption: 'keyModifierOption',
+    keyModifierOptionCommand: 'keyModifierOptionCommand',
+    keyModifierShift: 'keyModifierShift',
+    keyModifierShiftCommand: 'keyModifierShiftCommand',
+};
+
+/**
+ * Gets modifiers from a keyboard event.
+ *
+ * @param {Event} event
+ * @returns {Array<String>}
+ */
+function getKeyEventModifiers(event) {
+    const modifiers = [];
+    if (event.shiftKey) {
+        return constants.keyModifierShift;
+    }
+    if (event.ctrlKey) {
+        return constants.keyModifierControl;
+    }
+    if (event.altKey) {
+        return constants.keyModifierOption;
+    }
+    if (event.metaKey) {
+        return constants.keyModifierCommand;
+    }
+    return modifiers;
+}
+
+/**
+ * Finds index of register key command and listen to the event.
+ *
+ * @param {Object} json - List of key command objects.
+ * @param {string} json.input - any character key from the keyboard.
+ * @param {number} json.modifierFlags - predefined command from getConstants enum.
+ *
+ * @returns {number} index of registered keyCommand.
+ */
+function registeredCommandIndex(json) {
+    return _.findIndex(commands, item => item.input === json.input && item.modifierFlags === json.modifierFlags);
+}
+
+/**
+ * Checks if event exists in registered key command array and returns index.
+ *
+ * @param {Event} event
+ * @returns {number} index of registered keyCommand.
+ */
+function eventMatchesInput(event) {
+    const modifierFlags = getKeyEventModifiers(event);
+    return registeredCommandIndex({input: event.key.toLowerCase(), modifierFlags});
+}
+
+function onKeyDown(event) {
+    const index = eventMatchesInput(event);
+    if (index === -1) {
+        return;
+    }
+
+    EventEmitter.emit('onKeyCommand', commands[index]);
+}
+
+function getConstants() {
+    return constants;
+}
+
+document.removeEventListener('keydown', onKeyDown, {capture: true});
+document.addEventListener('keydown', onKeyDown, {capture: true});
```

Comment:
It seems the code changed after you added comments. the new position will be before the exports.

Reply:
+1

Reply:
@azimgd just making sure that you address this comment

Reply:
+1

Reply:
updated